actionmailer/lib/action_mailer/log_subscriber.rb

```diff
@@ -6,9 +6,16 @@ module ActionMailer
   class LogSubscriber < ActiveSupport::LogSubscriber
     # An email was delivered.
     def deliver(event)
-      info do
-        recipients = Array(event.payload[:to]).join(', ')
-        "\nSent mail to #{recipients} (#{event.duration.round(1)}ms)"
+      recipients = Array(event.payload[:to]).join(', ')
+
+      if exception = event.payload[:exception]
+        error do
+          "\n#{exception.first} was raised when sending mail to #{recipients} (#{duration_in_ms(event)})"
```

Comment:
The comments on `ActiveSupport::Notifications` seems to confirm that

```
  # If an exception happens during that particular instrumentation the payload will
  # have a key <tt>:exception</tt> with an array of two elements as value: a string with
  # the name of the exception class, and the exception message.
```

there's also a test for it..

ref:
- https://github.com/ignatiusreza/rails/blob/action_mailer_log_subscriber_on_exception/activesupport/lib/active_support/notifications.rb#L64-L66
- https://github.com/ignatiusreza/rails/blob/action_mailer_log_subscriber_on_exception/activesupport/test/notifications_test.rb#L209-L221